godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/type/array/core/AABBArray.kt

```diff
@@ -0,0 +1,35 @@
+package godot.core
+
+import godot.gdnative.godot_array
+import kotlinx.cinterop.COpaquePointer
+import kotlinx.cinterop.CValue
+
+@ExperimentalUnsignedTypes
+class AABBArray : CoreArray<AABB> {
+    constructor() : super()
+    constructor(other: AABBArray) : super(other)
+    internal constructor(native: CValue<godot_array>) : super(native)
+    internal constructor(mem: COpaquePointer) : super(mem)
+
+    override fun getCore(value: Variant): AABB = value.asAABB()
+    override fun getCoreArray(value: CValue<godot_array>) =
+        AABBArray(value)
+}
+
+@ExperimentalUnsignedTypes
+fun AABBArrayOf(vararg elements: AABB) = AABBArray().also {
+    for (arg in elements) {
+        it.append(arg)
+    }
+}
+
+@ExperimentalUnsignedTypes
+fun Iterable<AABB>.toVariantArray() = AABBArray().also{
```

Comment:
As we discussed. This function will be a performance problem. I suggest we add a kDoc to all the `.toVariantArray()` functions stating this does not perform well and if the user wants good performance he has to use variantArray directly.